### PR TITLE
[REVIEW] Moved `table.hpp` and rename `src/dataframe`

### PR DESCRIPTION
**Pull request by jrhemstad**
No functional changes. Purely organizational changes. 

- [x] Moved `column.cpp` and `context.cpp` to new `column/` directory
- [x] Renamed `src/dataframe` to `src/table`
- [x] Moved `table.hpp` class out of `cpp/include` to new `src/table/` directory
- [x] Updates `types.hpp` to be simple type declarations only
- [x] Updated includes for new directory structure

### CODE REVIEW

File: cpp/include/types.hpp [hunk 1]
```diff
@@ -13,171 +13,16 @@
...
-  gdf_size_type _num_rows{0};           ///< The number of elements in each column
-};
-
-}  // namespace cudf
-
-#endif
+// Forward declaration
+namespace cudf {
+struct table;
+}
```
Comment (harrism) [inline cpp/include/types.hpp]:
```suggestion
}

```

File: cpp/src/table/table.hpp [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+
+/**
+ * @brief A wrapper for a set of gdf_columns of equal number of rows.
+ *
+ */
+struct table {
+  /**---------------------------------------------------------------------------*
+   * @brief Constructs a table object from an array of `gdf_column`s
+   *
+   * @param cols The array of _columns wrapped by the table
```
Comment (harrism) [inline cpp/src/table/table.hpp]: Why the underscore?

```suggestion
   * @param cols The array of columns wrapped by the table
```

File: cpp/src/table/table.hpp [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+   * @param num_cols  The number of _columns in the array
```
Comment (harrism) [inline cpp/src/table/table.hpp]:
```suggestion
   * @param num_cols  The number of columns in the array
```

File: cpp/src/table/table.hpp (hunk 2 above)
Comment (jrhemstad) [reply cpp/src/table/table.hpp]: Replace all :smile:

**Pull request merged**